File: problem/Count_MaxOrSubsets/countMaxOrSubsets.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var countMaxOrSubsets = function(nums) {
    let maxOr = 0;
    for (let n of nums) maxOr |= n;

    function dfs(index, currOr) {
        if (index === nums.length) return currOr === maxOr ? 1 : 0;
        let include = dfs(index + 1, currOr | nums[index]);
        let exclude = dfs(index + 1, currOr);
        return include + exclude;
    }

    return dfs(0, 0);
};

File: problem/Count_MaxOrSubsets/countMaxOrSubsets.py
class Solution:
    def countMaxOrSubsets(self, nums):
        max_or = 0
        for n in nums:
            max_or |= n

        def dfs(i, curr_or):
            if i == len(nums):
                return 1 if curr_or == max_or else 0
            include = dfs(i + 1, curr_or | nums[i])
            exclude = dfs(i + 1, curr_or)
            return include + exclude

        return dfs(0, 0)